src/raspicam_node.cpp

```diff
@@ -1305,21 +1305,25 @@ void reconfigure_callback(raspicam_node::CameraConfig& config, uint32_t level, R
 
 int main(int argc, char** argv) {
   ros::init(argc, argv, "raspicam_node");
-  ros::NodeHandle n("~");
+  ros::NodeHandle nh_params("~");
+
+  bool private_topics;
+  nh_params.param<bool>("private_topics", private_topics, true);
+  ros::NodeHandle nh_topics(private_topics ? std::string("~") : std::string());

```

Comment:
Add a comment explaining that this line is using a private namespace or global one based on the parameter.